🔥 remove the mentioned of locale from quilt_rails and link the exampl… (#1300)
Shopify/quilt: gems/quilt_rails/README.md

#### Customizing the node server

> By default, sewing-kit bundles in [`@shopify/react-server-webpack-plugin`](../../packages/react-server-webpack-plugin/README.md) for `quilt_rails` applications to get apps up and running fast without needing to manually write any node server code. If what it provides is not sufficient, a completely custom server can be defined by adding a `server.js` or `server.ts` file to the app folder.

Reviewer: Love it, very clear.

      └─- server.{js|ts}x
```

> The simplest way to build a custom server that will work with this library is to use the APIs provided by [`@shopify/react-server`](../../packages/react-server/README.md).

Reviewer: ```suggestion
The simplest way to customize the server is to create a `server.tsx` file in your `app/ui` folder. This server file should export a [`@shopify/react-server`](../../packages/react-server/README.md) `createServer` call.
```

      └─- server.{js|ts}x
```

Use [`@shopify/react-server`](../../packages/react-server/README.md) to create a simple custom node server.

> Alternatively, if no customization on the server is needed, [`@shopify/react-server-webpack-plugin`](../../packages/react-server-webpack-plugin/README.md) can be use to have a virtual server created for you.

Author: Move to linking the react-server doc. So the example it reference will always be up to date.

Reviewer: Thanks for updating the docs on this so quickly! Two thoughts:
- I find the instructions here a bit confusing. Won't quilt_rails set up the node server for you? Therefor, if you don't need to customize it you won't need to do anything right?
- Secondly, I feel like a missing piece here is why you might want to customize it. In my case I'm trying to get the locale to pass it into the i18n manager which doesn't seem to be in there by default. I feel like for common use-cases such as that we should have some kind of docs that explain how to do that.

Author: * You are right. `quilt_rails` itself does not give you a node server, but by using `sewing-kit` it will automatically install `@shopify/react-server-webpack-plugin` and use it for you. So it does seem automatic from the outside, I just completely forgot about it 😬 

* I like the idea of doing common use-cases. Locale is not one thou as I think you can do it without. I will do some research but the only real need for customization I know of is web doing crazy auth things. But they dont use `quilt_rails`
